meta-balena-common/recipes-support/resin-init/resin-init-flasher/resin-init-flasher

```diff
@@ -192,18 +193,27 @@ IMAGE_FILE_SIZE=$(wc -c /opt/"$BALENA_IMAGE" | awk '{print $1}')
 
 # Check for UEFI mode and set variables accordingly.
 if [ -d /sys/firmware/efi ]; then
+    SECUREBOOT_ENABLED=$(
+        [ "$(jq .installer.secureboot "${CONFIG_PATH}")" = "true" ] \
+        && echo 1)
     SECUREBOOT_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SecureBoot \
         | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
     SETUPMODE_VAR=$(efivar -p -n 8be4df61-93ca-11d2-aa0d-00e098032b8c-SetupMode \
         | awk 'NR==1, $1 == "Value:" {next}; NF {print $2}')
 
-    if [ "${SECUREBOOT_VAR##+(0)}" = "1" ]; then
-        LUKS=1
+    if [ "${SECUREBOOT_ENABLED}" = "1" ]; then
+        if [ "${SECUREBOOT_VAR##+(0)}" = "1" ]; then
+            fail "Secure boot is configured, but is not supported in firmware."
```

Comment:
@jakogut could you please make sure that in setupmode secureboot is also always enabled? The code before this change did not assume they both had to be enabled, so it would work in setup mode independently of the secure boot configuration.